activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -1032,9 +1033,12 @@ def extract_foreign_key_action(structure, name, action) # :nodoc:
         end
       end
 
+      def create_table_info_cache # :nodoc:
+        @create_table_info_cache ||= {}
```

Comment:
Ah, it is only used in the schema dump, so we don't need to worry about thread-safety.